crates/core/src/factories/core_space.rs

```diff
@@ -450,7 +450,22 @@ impl Space for CoreSpace {
                 }
 
                 // also send the tombstone to the bootstrap server
-                self.bootstrap.put(info);
+                self.bootstrap.put(info.clone());
+
+                // and send it to our peers.
+                if let Err(err) = broadcast_agent_info(
```

Comment:
Is this really what we want here? It's getting *all* agents from the peer_store, and putting this in the publish queue. At least for the last agent, if the space gets shut down right after, the publish module won't have a chance to actually publish it.

Reply:
No, not really. It would be better to:
- Only select peers we have an open connection with, we don't want to be opening new connections at this point
- Do this send immediately, with no channel, so that when this method completes there is no more work to do and it's safe to shut down the space.

I can work on that, but this is one of several places where we're going to wide and if I'm going to start solving problems like that then I think we want to start adding tests for it and for them to pass, we need to fix other places that cause us to connect to everyone. E.g. gossip, and the other use of this `broadcast_agent_info` method, and probably others.

Or I can do this change as a best effort to get it right so it's one less thing to tune later?

Reply:
I'm cool merging as-is. It's just another item that will need to be addressed before we can scale past a handful of peers.

Reply:
Agreed, yes it is. I'm quite tempted to start tackling that problem before Holochain is ready for it... but it will only really work in Holochain when `holochain_p2p` respects it too and it will be difficult to prevent applications breaking that logic.